src/coroutine/memoset/prove.rs

```diff
@@ -0,0 +1,354 @@
+use crate::{
+    circuit::gadgets::pointer::AllocatedPtr,
+    error::ProofError,
+    field::LurkField,
+    lem::{pointers::Ptr, store::Store},
+    proof::{
+        nova::{CurveCycleEquipped, E1},
+        supernova::{Proof, PublicParams, SuperNovaPublicParams, C2, SS1, SS2},
+        FrameLike, Prover, RecursiveSNARKTrait,
+    },
+};
+
+use bellpepper_core::{num::AllocatedNum, ConstraintSystem, SynthesisError};
+use once_cell::sync::OnceCell;
+use tracing::info;
+
+use super::query::Query;
+use super::{
+    CoroutineCircuit, LogMemo, MemoSet, Scope, DEFAULT_RC_FOR_QUERY,
+    DEFAULT_TRANSCRIBE_INTERNAL_INSERTIONS,
+};
+
+use nova::{
+    supernova::{
+        error::SuperNovaError, snark::CompressedSNARK, NonUniformCircuit, RecursiveSNARK,
+        StepCircuit,
+    },
+    traits::{
+        snark::{BatchedRelaxedR1CSSNARKTrait, RelaxedR1CSSNARKTrait},
+        Dual as DualEng,
+    },
+};
+use std::marker::PhantomData;
+
+const COROUTINE_ARITY: usize = 12;
+
+type Coroutine<'a, F, Q> = CoroutineCircuit<'a, F, LogMemo<F>, Q>;
+
+#[derive(Debug)]
+pub(crate) struct MemosetProver<'a, F, Q> {
+    /// The number of small-step reductions performed in each recursive step of
+    /// the primary Lurk circuit.
```

Comment:
Is this comment copy-pasted from elsewhere? I don't think it describes the use of `reduction_count` here.

Reply:
I think I've copy pasted indeed